runtime/near-vm-runner/src/cache.rs

```diff
@@ -292,3 +292,61 @@ pub mod wasmer1_cache {
         }
     }
 }
+
+pub(crate) fn precompile_contract_impl(
+    vm_kind: VMKind,
+    wasm_code: &ContractCode,
+    config: &VMConfig,
+    cache: Option<&dyn CompiledContractCache>,
+) -> Result<bool, ContractPrecompilatonError> {
+    let cache = match cache {
+        None => return Ok(false),
+        Some(it) => it,
+    };
+    let key = get_key(wasm_code, vm_kind, config);
+    // Check if we already cached with such a key.
+    match cache.get(&(key.0).0) {
```

Comment:
what is the type of `key`? `(key.0).0` doesn't look very nice

Reply:
We need to get raw bits of the CryptoHash. Any better way?

Reply:
`impl CryptoHash { fn raw_bytes(&self) -> _ { self.0 }` ?

Reply:
Ok, added

Reply:
couldn't you do `.as_ref`?

Reply:
Not really, as some use cases need exactly sized array, i.e. ` SerializationError { hash: [u8; 32] }` and `as_ref` returns `&[u8]`. It may be worth reworking, but looks like something not belonging to this PR.